src/script_number.js

```diff
@@ -0,0 +1,78 @@
+function decode (buffer, maxLength, minimal) {
+  maxLength = maxLength || 4
+  minimal = minimal === undefined ? true : minimal
+
+  var length = buffer.length
+  if (length === 0) return 0
+  if (length > maxLength) throw new TypeError('Script number overflow')
+  if (minimal) {
+    if ((buffer[length - 1] & 0x7f) === 0) {
+      if (length <= 1 || (buffer[length - 2] & 0x80) === 0) throw new Error('Non-minimally encoded script number')
+    }
+  }
+
+  var result
+
```

Comment:
@fanatid never mind, works fine, will have a look at it again tomorrow.
To avoid the 40-bit special case, I'd need a non-integer equivalent of the negative handling code (aka, `-result` on `0x80`).